lib/mongoose_push/service/apns/certificate.ex

```diff
@@ -0,0 +1,46 @@
+defmodule MongoosePush.Service.APNS.Certificate do
+  @moduledoc false
+
+  require Record
+  Record.defrecord :otp_cert,
+    Record.extract(:OTPCertificate, from_lib: "public_key/include/public_key.hrl")
+  Record.defrecord :tbs_cert,
+    Record.extract(:TBSCertificate, from_lib: "public_key/include/public_key.hrl")
+  Record.defrecord :cert_ext,
+    Record.extract(:Extension,      from_lib: "public_key/include/public_key.hrl")
+
+  # From: https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CommunicatingwithAPNs.html
+  @apns_topic_extn_id {1, 2, 840, 113635, 100, 6, 3, 6}
+
+  def extract_topics!(cert_file) do
+    maybe_extension =
+      cert_file
+      |> File.read!()
+      |> :public_key.pem_decode()
+      |> List.keyfind(:Certificate, 0)
+      |> get_cert_extension!(@apns_topic_extn_id)
+
+    if maybe_extension == nil do
+        throw :no_extension
```

Comment:
Well, the issue here is that this whole code can crash due to... well, reasons. There is one catch for that, only in order to log the issue properly. Here I throw `:no_extension` only because badmatch on `nil` is not saying anything useful. This is not an error condition that changes logic somehow, at least not in different way then any other `error` here. This is just to make this one error more readable in logs.